os/pm/pm_procfs.c

```diff
@@ -586,35 +615,25 @@ static int power_readdir(struct fs_dirent_s *dir)
 	/* Have we reached the end of the directory */
 	index = powerdir->base.index;
 	if (index >= powerdir->base.nentries) {
-		fdbg("Entry %d: End of directory\n", index);
 		return -ENOENT;
 	}
-
-	if (powerdir->base.level < 1 || powerdir->base.level > 3) {
+	if (powerdir->base.level < 1 || powerdir->base.level >= 3) {
 		fdbg("Invalid directory level\n");
 		return -ENOENT;
 	}
-
 	if (powerdir->base.level == 1) {
 		/* Listing the contents of domains */
-		if (index >= 0 && index < CONFIG_PM_NDOMAINS) {
+		if (index >= 0 && index < g_pmglobals.ndomains) {
 			dir->fd_dir.d_type = DTYPE_DIRECTORY;
 			snprintf(dir->fd_dir.d_name, sizeof(dir->fd_dir.d_name), "%d", index);
```

Comment:
Could you please change index to domain name?

Reply:
Done